library/src/main/java/com/bumptech/glide/GlideBuilder.java

```diff
@@ -292,10 +306,16 @@ GlideBuilder setEngine(Engine engine) {
 
   public Glide build(Context context) {
     if (sourceExecutor == null) {
+      if (uncaughtThrowableStrategy != null) {
+        sourceExecutor = GlideExecutor.newSourceExecutor(uncaughtThrowableStrategy);
+      }
       sourceExecutor = GlideExecutor.newSourceExecutor();
     }
 
     if (diskCacheExecutor == null) {
+      if (uncaughtThrowableStrategy != null) {
+        diskCacheExecutor = GlideExecutor.newDiskCacheExecutor(uncaughtThrowableStrategy);
+      }
```

Comment:
It looks like you're overriding this just below with newDIskCacheExecutor() always? same for newSourceExecutor above.